proposals/0721-golden-template-crnl.md

```diff
@@ -0,0 +1,178 @@
+
+### **Documenting the supported React Native versions**
+
+Currently, there is no such section inside the [create-react-native-library repository](https://github.com/callstack/react-native-builder-bob)’s documentation to list the compatible React Native versions. To address this issue, supported versions can be tested manually.
+
+### Versioning `create-react-native-library`
+
+`create-react-native-library` will follow a versioning schema based on **`major.minor.patch`** as outlined below:
+
+1. The **patch** version will be incremented for bug fixes.
+2. The **minor** version will be incremented for updates to dependencies and the introduction of new features in the template. **Minor** versions will not remove support for any React Native versions.
+3. The **major** version will be incremented for breaking changes, such as dropping support for specific React Native versions.
+
+### **Minimum supported versions of iOS and Android**
+
+The library template aims to support the same minimum versions of host operating systems as the app template. This translates to `13.4` on iOS and `6.0` on Android.
+
+---
+
+> The following points are possible additions to the proposal. They can be added in the future.
+> 
+
+### Using the [React Native Test App](https://github.com/microsoft/react-native-test-app) as the example app (possible addition)
+
+> “React Native Test App (RNTA) provides test apps for all platforms as a package. It handles the native bits for you so you can focus on what's important: your product.” [-React Native Test App’s readme](https://github.com/microsoft/react-native-test-app/blob/trunk/README.md)
+> 
+
+React Native Test App can be used to test the libraries. It’s easier to maintain than an app created with the app template since it handles the native parts.
+
+### Reducing the amount of boilerplate code in the template (possible addition)
+
+As many libraries share some common logic, the proposal suggests moving some of the boilerplate code to the core template, simplifying individual library templates. You can take the [build.gradle file of a native module](https://github.com/atlj/official-template-proposal-module/blob/main/android/build.gradle) as an example. There are a lot of lines that are shared between many libraries.
+
+---
+
+## **Drawbacks**
+
+The primary drawback of this proposal is the additional maintenance cost for the core React Native repository. The React Native team needs to actively monitor if their changes break the template, fix any regressions, and communicate the necessary changes to the template.
+
+## **Alternatives**
+
+Various alternatives to `create-react-native-library` exist, listed here in no particular order:
+
+- [https://github.com/brodybits/react-native-module-init (not maintained)](https://github.com/brodybits/react-native-module-init)
+- [https://github.com/brodybits/create-react-native-module (not maintained)](https://github.com/brodybits/create-react-native-module)
+
```

Comment:
Would it be appropriate to mention `create-expo-module` here?

Reply:
Although it's a similar tool, it doesn't have the exact same functionality as CRNL. AFAIK you cannot create a native library that's compatible with a Bare React Native app using create-expo-module.

Reply:
I suppose not by itself.

As far as I know, the only extra thing you should need to do to use it in a bare RN app is `npx install-expo-modules@latest`

Reply:
https://docs.expo.dev/bare/installing-expo-modules/

Reply:
It has a different API than the API of native modules and views in React Native, so it cannot be considered an alternative CLI to create them. It's an alternative way to write modules which is not what this section is meant for.